Resources/public/js/apps/plugins/ez-languageselectionboxplugin.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-languageselectionboxplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the language selection box plugin
+     *
+     * @module ez-languageselectionboxplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * The language selection box plugin. It's a plugin the application to set up the
+     * `languageSelectionBoxOpen` and `languageSelectionBoxClose` event handlers.
+     *
+     * @namespace eZ.Plugin
+     * @class LanguageSelectionBox
+     * @constructor
+     * @extends Plugin.Base
+     */
+    Y.eZ.Plugin.LanguageSelectionBox = Y.Base.create('languageSelectionBoxPlugin', Y.Plugin.Base, [], {
+        initializer: function () {
+            var app = this.get('host');
+
+            app.on('*:languageSelectionBoxOpen', function (e) {
```

Comment:
I would rather use more meaningful event names (modeled after the ones used in the universal discovery widget instead) like: `languageSelect`, `languageSelected` and `cancelLanguageSelect`